examples/bollinger/bollinger/jobs/bollinger_analysis.py

```diff
@@ -0,0 +1,43 @@
+# pylint: disable=redefined-outer-name
+from dagster import asset, build_assets_job
+
+from ..lib import (
+    AnomalousEventsDgType,
+    BollingerBandsDgType,
+    StockPricesDgType,
+    compute_anomalous_events,
+    compute_bollinger_bands_multi,
+    load_sp500_prices,
+)
+
+@asset(
+    dagster_type=StockPricesDgType,
+    metadata={"owner": "alice@example.com"},
+)
+def sp500_prices():
+    """Historical stock prices for the S&P 500."""
+    return load_sp500_prices()
```

Comment:
Thoughts on inlining these functions? It would help users avoid the impression that they need to have a layer of indirection for every asset definition.

Reply:
The reason they aren't inlined is because there is an accompanying notebook uses them-- I can inline them if you want, but I think having the code factored out so that it can be used elsewhere (like a notebook) illustrates a pattern a significant number of users will use.

Reply:
Similar to ops, I think that ideally assets would be directly invokable, and then the notebook could just invoke the asset directly. Thoughts? Feel free to merge without addressing this.

Reply:
Hmm-- maybe a matter of style but personally, calling asset (or op) definitions in non-Dagster code feels very unintuitive to me. When I define an op or asset, I am wrapping an underlying computation with a bunch of metadata. If I'm using that computation in multiple contexts (e.g. Dagster jobs AND exploratory notebooks), it feels more natural to have the core computation that remains constant factored out into plain old Python functions, and layer context-specific metadata on top of that, rather than have my "source of truth" for the logic be tied to framework-specific abstractions, especially when IMO there's not a clear benefit as a user (illustrative benefits for a Dagster example aside) from doing so. 

Like, what if I want to pass my functions to some other framework (like an ML framework) that does some inspection of the functions parameters and so forth rather than just invoking `__call__`? Not sure if that would work, but as a user I'm rarely 100% sure what a framework decorator is doing, so I would feel uncomfortable just slinging around ops/assets like this.